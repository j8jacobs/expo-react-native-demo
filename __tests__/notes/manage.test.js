jest.mock("expo-router", () => {
  return {
    router: {
      push: jest.fn(),
    },
  };
});

import ManageNotes from "../../app/notes/manage";
import renderer from "react-test-renderer";
import {
  render,
  screen,
  fireEvent,
  waitFor,
} from "@testing-library/react-native";
import { renderRouter, ExpoRouterscreen } from "expo-router/testing-library";
import { View, Text } from "react-native";
import { router } from "expo-router"; // This is how you'd use router in your component, not in your test

describe("<ManageNotes />", () => {
  it("should render", () => {
    render(<ManageNotes />);
  });
  it("should render correctly", () => {
    const tree = renderer.create(<ManageNotes />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should route to create page when btn is pressed", async () => {
    const MockComponent = jest.fn(() => <View />);
    renderRouter(
      {
        "/notes/manage": ManageNotes,
        "/notes/create": MockComponent,
      },
      {
        initialUrl: "/notes/manage",
      }
    );

    render(<ManageNotes />);
    await waitFor(() => {
      console.log("waiting for...");
      expect(screen.getByText("Add Note")).toBeTruthy();
      console.log("done");
    });
    fireEvent.press(screen.getByText("Add Note"));
    expect(router.push).toHaveBeenCalledWith("/notes/create");
  });
});
