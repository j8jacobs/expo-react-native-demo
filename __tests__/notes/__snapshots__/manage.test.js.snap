// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageNotes /> should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
      "position": "relative",
    }
  }
>
  <Text>
    Manage Notes Page
  </Text>
  <View />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#EE8434",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 24,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      Add Note
    </Text>
  </View>
</View>
`;
